SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_MaterialFarmer.cpp

```diff
@@ -0,0 +1,566 @@
+
+    // Jump
+    pbf_press_button(context, BUTTON_B, 125, 30);
+
+    // Fly 
+    pbf_press_button(context, BUTTON_B, 50, 10); 
+    pbf_press_button(context, BUTTON_B, 50, 10); // Double click in case of drop
+    pbf_press_button(context, BUTTON_LCLICK, 50, 0);
+
+    // you automatically move forward  when flying without pressing any buttons. 
+    // so, just wait.
+    pbf_wait(context, 2200);
+
+    // arrived at start position. stop flying
+    pbf_press_button(context, BUTTON_B, 50, 400);
+    // get off ride
+    pbf_press_button(context, BUTTON_PLUS, 50, 50);
+
+    // extra B presses to ensure we stop flying, in case the previous B press
+    // was dropped. This way, you eventually reset back to Pokecenter, instead
+    // of flying until an exception is thrown.
+    pbf_press_button(context, BUTTON_B, 50, 10);
+    pbf_press_button(context, BUTTON_B, 50, 10);
+
+    // look right
+    pbf_move_right_joystick(context, 255, 128, 20, 10);
+
+    // move forward slightly
+    pbf_move_left_joystick(context, 128, 0, 50, 10);
+
+    env.console.log("Arrived at Let's go start position", COLOR_PURPLE);
+}
+
+
+// wait, then move forward quickly
+void MaterialFarmer::lets_go_movement0(BotBaseContext& context){
+    pbf_wait(context, 500);
+    pbf_move_left_joystick(context, 128, 0, 200, 10);
+}
+
+// wait, then move forward quickly, then wait some more.
+void MaterialFarmer::lets_go_movement1(BotBaseContext& context){
+    pbf_wait(context, 500);
+    pbf_move_left_joystick(context, 128, 0, 100, 10);
+    pbf_wait(context, 100);
+}
+
+
+// One iteration of the hunt: 
+// start at North Province (Area 3) pokecenter, go out and use Let's Go to battle , 
+void MaterialFarmer::run_lets_go_iteration(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    auto& console = env.console;
+    // - Orient camera to look at same direction as player character
+    // - This is needed because when save-load the game, the camera points
+    // in the same direction as the player.
+    // - But when warping to pokecenter, the camera is facing the player.
+    pbf_press_button(context, BUTTON_L, 50, 40);
+
+    // zoom out camera
+    pbf_move_right_joystick(context, 128, 255, 45, 10);
+
+    const bool throw_ball_if_bubble = false;
+    const int total_iterations = 13;
+
+    auto move_forward_with_lets_go = [&](int num_iterations){
+        context.wait_for_all_requests();
+        for(int i = 0; i < num_iterations; i++){
+            use_lets_go_to_clear_in_front(console, context, *m_encounter_tracker, throw_ball_if_bubble, [&](BotBaseContext& context){
+                // Do the following movement while the Let's Go pokemon clearing wild pokemon.
+                env.console.log("Move-forward iteration number: " + std::to_string(i + 1) + "/" + std::to_string(total_iterations), COLOR_PURPLE);
+
+                lets_go_movement1(context);
+            });
+        }
+    };
+
+    move_forward_with_lets_go(total_iterations);
```

Comment:
Why split this into lambda instead of just inlining the code?

Reply:
Now that you mention it, I could. It's like this because I based this off of the Scatterbug program, where a lambda is needed.